packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -237,6 +238,14 @@ export type TableProps<
    * @beta
    */
   enableVirtualization?: boolean;
+  /**
+   * Initialize the table scrolled to this item.
+   *
+   * Must be used in conjunction with `getRowId` prop.
+   *
+   * Does not support tables with pagination.
+   */
+  scrollToItem: T;
```

Comment:
This should be an optional prop.

Reply:
If we require users to pass `getRowId` maybe instead of passing data item, they should pass id? I think by default `getRowId` returns row index so in theory users don't even need to pass `getRowId` if they are ok to pass index.

Reply:
Yes, it should be optional 😁. Updated that.

The reason I'm using both `getRowId` and `item: T` is that we can run `getRowId(item)` and compare that to the `getRowId` on each of the rows to find the exact row being referenced. We could ask the user to pass in the row Id for the row, but it seems easier not to.

I had originally implemented by checking `row.original === item`, but I had some cases in the app I'm using where the behind the scenes data-loading mutated some info on the item, and the equality check didn't work. I figure a way to get an unmutable id for the item is safer.